examples/nemo/eg4/tra_adv_compute.F90

```diff
@@ -38,7 +38,7 @@ subroutine tra_adv_compute(pun, pvn, pwn, umask, vmask, tmask, mydomain, jpi, jp
     !END DO
 
     ! Not sure how to cope with implicit loops in the SIR backend
-    ! zwx(:,:,jpk) = 0.e0   ;   zwy(:,:,jpk) = 0.e0
+    zwx(:,:,jpk) = 0.e0   ;   zwy(:,:,jpk) = 0.e0
```

Comment:
Whoop! (But can we now remove the comment at line 40?)

Reply:
Actually you'll have to hold the whoop as that should not be un-commented yet. I accidentally left it when testing. The problem is that the front end still creates nemoimplicitloop and the sir back-end falls over as it does not support it. This will be fixed in PR #829, so I'll uncomment the code and add the transformation to the relevant script in that PR. Apologies.

Reply:
Sorry if I've caused confusion. The nemo frontend currently creates a nemoimplicitloop when it sees Fortran array notation. Therefore the transformation has no effect on the code as it looks for arrays with ranges. As the SIR back-end does not support nemoimplicitloop it falls over. When we remove the creation of nemoimplicit loop the nemo front end will create arrays with ranges, the transformation will change these to loops and the SIR back-end will generate code. Actually that is not quite true as the SIR backend does not support lbound and ubound so we might need to wait for #815 and hope the bounds are available (otherwise we will have to make some assumptions about bounds). Does that help explain?